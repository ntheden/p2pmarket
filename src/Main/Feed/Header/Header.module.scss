.header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    background: white;

    .username {
        margin-left: 1em;
    }

    .threeDots {
        justify-content: flex-end;
    }
}

:global(.list-group-item) {
    text-align: center;
    a {
        text-decoration: none;

        &.markedLink {
            color: red;
            font-weight: bold;
        }
    }
}
