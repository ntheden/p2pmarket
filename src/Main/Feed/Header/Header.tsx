import React, { useState } from 'react';

import { Button, Card, Image, ListGroup, Modal } from 'react-bootstrap';
import { BsThreeDots } from 'react-icons/bs';

import { Username } from 'lib/ui/Username/Username';

import styles from './Header.module.scss';

export const Header = () => {
    const [postMenuVisibility, setPostMenuVisibility] = useState(false);

    return (
        <Card.Header className={styles.header}>
            <div>
                <Image
                    className="w-32"
                    src="https://picsum.photos/id/101/32/32"
                    roundedCircle={true}
                    thumbnail={true}
                />
                <Username username="azizoid" className={styles.username} />
            </div>
            <Card.Link className={styles.threeDots}>
                <Button
                    onClick={() => setPostMenuVisibility((prev) => !prev)}
                    variant="link">
                    <BsThreeDots role="icon" />
                </Button>
                <Modal
                    show={postMenuVisibility}
                    onHide={() => setPostMenuVisibility(false)}
                    centered={true}>
                    <Modal.Body>
                        <ListGroup variant="flush">
                            {[
                                { title: 'Report', mark: true, href: '#' },
                                { title: 'Unfollow', mark: true, href: '#' },
                                { title: 'Go to post', mark: false, href: '#' },
                                {
                                    title: 'Share to...',
                                    mark: false,
                                    href: '#',
                                },
                                { title: 'Copy link', mark: false, href: '#' },
                                { title: 'Embed', mark: false, href: '#' },
                            ].map(({ title, mark, href }, index) => (
                                <ListGroup.Item key={index}>
                                    <a
                                        href={href}
                                        {...(mark && {
                                            className: styles.markedLink,
                                        })}>
                                        {title}
                                    </a>
                                </ListGroup.Item>
                            ))}
                            <ListGroup.Item>
                                <Button
                                    onClick={() => setPostMenuVisibility(false)}
                                    variant="link"
                                    style={{ textDecoration: 'none' }}>
                                    Close
                                </Button>
                            </ListGroup.Item>
                        </ListGroup>
                    </Modal.Body>
                </Modal>
            </Card.Link>
        </Card.Header>
    );
};
