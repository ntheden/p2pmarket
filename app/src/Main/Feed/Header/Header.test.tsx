import React from 'react';
import { render } from '@testing-library/react';
import { Header } from './Header';

test('renders Header', () => {
    const { container, getByText, getByRole } = render(<Header />);

    expect(getByRole('img')).toBeInTheDocument();
    expect(getByText('azizoid')).toBeInTheDocument();
    expect(getByRole('icon')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
});
