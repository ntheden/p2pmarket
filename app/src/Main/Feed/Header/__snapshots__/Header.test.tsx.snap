// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header 1`] = `
<div>
  <div
    class="header card-header"
  >
    <div>
      <img
        class="w-32 rounded-circle img-thumbnail"
        src="https://picsum.photos/id/101/32/32"
      />
      <a
        class="username username"
        href="/#username"
      >
        azizoid
      </a>
    </div>
    <a
      class="threeDots card-link"
    >
      <button
        class="btn btn-link"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          role="icon"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 9.5a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3zm5 0a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
          />
        </svg>
      </button>
    </a>
  </div>
</div>
`;
